go/ct/gen/code.go

```diff
@@ -114,12 +118,33 @@ func (g *CodeGenerator) Generate(assignment Assignment, rnd *rand.Rand) (*st.Cod
 		minSize += size
 	}
 
-	// We use an exponential distribution for the code size here since long codes
-	// extend the runtime but are expected to reveal limited extra code coverage.
-	const expectedSize float64 = 200
-	size := int(rnd.ExpFloat64()/(1/expectedSize)) + minSize
-	if size > st.MaxCodeSize {
-		size = st.MaxCodeSize
+	// Make enough space to host all the different opCodes used in condition.
+	opCount := make(map[OpCode]bool)
+	for _, constraint := range varOps {
+		opCount[constraint.op] = true
+	}
+	for _, constraint := range ops {
+		opCount[constraint.op] = true
+	}
+	minSize = max(minSize, len(opCount))
+
+	// If there are any variables that need to be bound to code, there must be at
+	// least one instruction in the resulting code.
+	if minSize == 0 && len(g.varIsCodeConstraints) > 0 {
+		minSize = 1
+	}
+
+	var size int
+	if g.codeSize != nil {
+		size = *g.codeSize + minSize
```

Comment:
This is surprising ... the field is named `codeSize` but it does not seem to put a constraint on the code size. Instead, it adds some extra "boost" to the minimum code size?

I would have thought to make it simply a strict constraint on the code size. If `codeSize` is set and it has a value, that is the size of the produced code. If the value is smaller than the `minSize`, the generator produces an `Unsatisfiable` error.

This way, the parameter has a simpler interpretation. Wdyt?

Reply:
Right, that would work if the code size would be generated using a range. but instead the generated size is using this ad-hoc code:
![image](https://github.com/Fantom-foundation/Tosca/assets/170023705/734d53ea-b8d0-4619-bba0-9e7127fe84d8)
I am not sure what takes precedence here, the feature of returning small code sizes seems reasonable to me. 
how would you proceed, would you add an exponential generator to the range generator tool?